ufo2fontir/src/toir.rs

```diff
@@ -114,28 +118,84 @@ where
 mod tests {
     use norad::designspace::DesignSpaceDocument;
     use ordered_float::OrderedFloat;
-    use std::path::Path;
+    use std::path::PathBuf;
+
+    use fontir::{coords::UserSpaceCoord, ir, piecewise_linear_map::PiecewiseLinearMap};
 
-    use crate::toir::designspace_to_ir;
-    use fontir::{coords::UserSpaceCoord, ir};
+    use crate::toir::to_ir_axis;
+
+    use pretty_assertions::assert_eq;
 
     fn user_coord(v: f32) -> UserSpaceCoord {
         UserSpaceCoord::new(OrderedFloat(v))
     }
 
+    fn testdata_dir() -> PathBuf {
+        PathBuf::from("../resources/testdata")
+    }
+
     #[test]
     fn simple_wght_variable() {
-        let ds = DesignSpaceDocument::load(Path::new("testdata/wght_var.designspace")).unwrap();
+        let ds = DesignSpaceDocument::load(testdata_dir().join("wght_var.designspace")).unwrap();
         assert_eq!(
             vec![ir::Axis {
                 name: "Weight".to_string(),
                 tag: "wght".to_string(),
                 min: user_coord(400_f32),
                 default: user_coord(400_f32),
                 max: user_coord(700_f32),
-                hidden: false
+                hidden: false,
+                design_to_user: PiecewiseLinearMap::nop(),
             }],
-            designspace_to_ir(ds).unwrap()
+            ds.axes.iter().map(to_ir_axis).collect::<Vec<_>>()
+        );
+    }
+
+    #[test]
+    fn simple_axis_mapping() {
+        let ds = DesignSpaceDocument::load(testdata_dir().join("mapping.designspace")).unwrap();
+        // Norad doesn't seem to give us the value from <labelname xml:lang="en">Weight</labelname> for axes so we get lowercase
+        assert_eq!(
+            vec![
+                ir::Axis {
+                    name: "weight".to_string(),
+                    tag: "wght".to_string(),
+                    min: user_coord(-1.0),
+                    default: user_coord(-0.1),
+                    max: user_coord(1.125),
+                    hidden: false,
+                    design_to_user: PiecewiseLinearMap::new(vec![
+                        (OrderedFloat(100.0), OrderedFloat(-1.0)),
+                        (OrderedFloat(200.0), OrderedFloat(-0.825)),
+                        (OrderedFloat(300.0), OrderedFloat(-0.55)),
+                        (OrderedFloat(400.0), OrderedFloat(-0.1)),
```

Comment:
you can see how the test designspace file has user-space coordinates in the "input" (or `from`), and design-space in the "output" (`to`)

Reply:
Done